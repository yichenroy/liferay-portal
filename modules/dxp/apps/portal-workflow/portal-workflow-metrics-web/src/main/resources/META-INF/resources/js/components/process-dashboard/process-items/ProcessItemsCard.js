import {AppContext} from '../../AppContext';
import Filter from '../../../shared/components/filter/Filter';
import Icon from '../../../shared/components/Icon';
import LoadingState from '../../../shared/components/loading/LoadingState';
import Panel from '../../../shared/components/Panel';
import PANELS from './Panels';
import React from 'react';
import ReloadButton from '../../../shared/components/list/ReloadButton';
import SummaryCard from './SummaryCard';
import Tooltip from '../../../shared/components/Tooltip';

export default class ProcessItemsCard extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			error: null,
			loading: false,
			process: {}
		};
	}

	componentWillMount() {
		return this.loadData();
	}

	componentWillReceiveProps(nextProps) {
		if (nextProps.timeRange !== this.props.timeRange) {
			return this.loadData(nextProps);
		}
	}

	loadData(props = this.props) {
		return this.requestData(props)
			.then(data => {
				this.context.setTitle(data.title);

				this.setState({
					error: null,
					loading: false,
					process: data
				});
			})
			.catch(() => {
				this.setState({
					error: Liferay.Language.get(
						'there-was-a-problem-retrieving-data-please-try-reloading-the-page'
					),
					loading: false
				});
			});
	}

	requestData(props = this.props) {
		const {client} = this.context;
		const {completed = false, processId, timeRange} = props;

		let urlRequest = `/processes/${processId}?completed=${completed}`;

		if (timeRange && Number.isInteger(timeRange.id)) {
			urlRequest += `&timeRange=${timeRange.id}`;
		}

		this.setState({
			loading: true
		});

		return client.get(urlRequest).then(({data}) => data);
	}

	render() {
		const {error, loading, process} = this.state;
		const {
			completed,
			description,
			filter,
			processId,
			timeRange,
			title
		} = this.props;

		const errorRender = Component =>
			(error && (
				<div className='pb-6 pt-5 text-center'>
					<p className='small'>{error}</p>
					<ReloadButton />
				</div>
			)) ||
			Component;

		const loadingRender = Component =>
			(loading && (
				<div className='pb-6 pt-5'>
					<LoadingState />
				</div>
			)) ||
			Component;

		return (
			<Panel>
				<Panel.Header
					elementClasses={[
						'dashboard-panel-header',
						filter && 'pb-0'
					]}
				>
					<div className='autofit-row'>
						<div className='autofit-col autofit-col-expand flex-row'>
							<span className='mr-3'>{title}</span>

							<Tooltip
								message={description}
								position='right'
								width='288'
							>
								<Icon iconName={'question-circle-full'} />
							</Tooltip>
						</div>

						{filter && (
							<div className='autofit-col m-0 management-bar management-bar-light navbar'>
								<ul className='navbar-nav'>
									<Filter
										{...filter}
										filterKey={filter.key}
										position='right'
									/>
								</ul>
							</div>
						)}
					</div>
				</Panel.Header>

				<Panel.Body>
					{errorRender(
						loadingRender(
							<div className={'pt-1 pb-4 d-flex'}>
								{PANELS.map((panel, index) => (
									<SummaryCard
										{...panel}
										completed={completed}
										key={index}
										processId={processId}
										timeRange={timeRange}
										total={
											panel.addressedToField ===
											panel.totalField
										}
										totalValue={process[panel.totalField]}
										value={process[panel.addressedToField]}
									/>
								))}
							</div>
						)
					)}
				</Panel.Body>
			</Panel>
		);
	}
}

ProcessItemsCard.contextType = AppContext;
