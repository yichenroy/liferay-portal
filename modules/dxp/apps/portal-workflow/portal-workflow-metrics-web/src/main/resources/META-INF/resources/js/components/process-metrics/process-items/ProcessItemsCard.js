import React, {useContext, useEffect, useState} from 'react';
import {AppContext} from '../../AppContext';
import {ErrorContext} from '../../../shared/components/request/Error';
import Icon from '../../../shared/components/Icon';
import {LoadingContext} from '../../../shared/components/request/Loading';
import Panel from '../../../shared/components/Panel';
import PANELS from './Panels';
import Request from '../../../shared/components/request/Request';
import SummaryCard from './SummaryCard';
import Tooltip from '../../../shared/components/Tooltip';

function ProcessItemsCard({
	children,
	completed,
	description,
	processId,
	timeRange,
	title
}) {
	return (
		<Request>
			<Panel>
				<ProcessItemsCard.Header
					children={children}
					description={description}
					title={title}
				/>

				<ProcessItemsCard.Body
					completed={completed}
					processId={processId}
					timeRange={timeRange}
				/>
			</Panel>
		</Request>
	);
}

ProcessItemsCard.Body = ({completed = false, processId, timeRange}) => {
	const {client, setTitle} = useContext(AppContext);
	const {setError} = useContext(ErrorContext);
	const {setLoading} = useContext(LoadingContext);

	const [process, setProcess] = useState(null);

	const fetchData = () => {
		setError(null);
		setLoading(true);

		let urlRequest = `/processes/${processId}?completed=${completed}`;

		if (timeRange && timeRange.dateEnd && timeRange.dateStart) {
			const {dateEnd, dateStart} = timeRange;

			urlRequest += `&dateEnd=${dateEnd.toISOString()}&dateStart=${dateStart.toISOString()}`;
		}

		return client
			.get(urlRequest)
			.then(({data}) => {
				setTitle(data.title);
				setProcess(data);
			})
			.catch(error => {
				setError(error);
			})
			.then(() => {
				setLoading(false);
			});
	};

	useEffect(() => {
		fetchData();
	}, [timeRange]);

	return (
		<Panel.Body>
			<Request.Error />

			<Request.Loading />

			<Request.Success>
				{process && (
					<div className={'pt-1 pb-4 d-flex'}>
						{PANELS.map((panel, index) => (
							<SummaryCard
								{...panel}
								completed={completed}
								key={index}
								processId={processId}
								timeRange={timeRange}
								total={
									panel.addressedToField === panel.totalField
								}
								totalValue={process[panel.totalField]}
								value={process[panel.addressedToField]}
							/>
						))}
					</div>
				)}
			</Request.Success>
		</Panel.Body>
	);
};

ProcessItemsCard.Header = ({children, description, title}) => (
	<Panel.Header
		elementClasses={['dashboard-panel-header', children && 'pb-0']}
	>
		<div className="autofit-row">
			<div className="autofit-col autofit-col-expand flex-row">
				<span className="mr-2">{title}</span>

				<Tooltip message={description} position="right" width="288">
					<Icon iconName={'question-circle-full'} />
				</Tooltip>
			</div>

			{children && (
				<Request.Success>
					<div className="autofit-col m-0 management-bar management-bar-light navbar">
						<ul className="navbar-nav">{children}</ul>
					</div>
				</Request.Success>
			)}
		</div>
	</Panel.Header>
);

export default ProcessItemsCard;
