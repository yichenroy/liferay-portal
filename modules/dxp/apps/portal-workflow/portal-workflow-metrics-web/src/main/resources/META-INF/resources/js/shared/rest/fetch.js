/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * The contents of this file are subject to the terms of the Liferay Enterprise
 * Subscription License ("License"). You may not use this file except in
 * compliance with the License. You can obtain a copy of the License by
 * contacting Liferay, Inc. See the License for the specific language governing
 * permissions and limitations under the License, including but not limited to
 * distribution rights of the Software.
 */

import axios from 'axios';

/**
 * @module fetch
 * @description Basic alias of fetch client to perform consults in portal rest API.
 * @example
 * import fetch from '@/shared/rest/fetch';
 * fetch.get('/process').then(res => console.log(res));
 */
const restClient = axios.create({
	baseURL: '/o/portal-workflow-metrics/v1.0'
});

axios.defaults.headers.common[
	'Accept-Language'
] = Liferay.ThemeDisplay.getBCP47LanguageId();

axios.defaults.params = {
	['p_auth']: Liferay.authToken
};

export default restClient;
