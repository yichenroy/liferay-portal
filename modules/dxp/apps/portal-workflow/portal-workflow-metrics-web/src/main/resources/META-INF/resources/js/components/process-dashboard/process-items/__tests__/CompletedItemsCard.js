import { CompletedItemsCard } from '../ProcessItemsCard';
import fetch from '../../../../test/mock/fetch';
import fetchFailure from '../../../../test/mock/fetchFailure';
import React from 'react';
import { MockRouter as Router } from '../../../../test/mock/MockRouter';

test('Should render component', () => {
	const data = {
		id: 35315,
		instanceCount: 30,
		onTimeInstanceCount: 20,
		overdueInstanceCount: 10,
		title: 'Single Approver'
	};

	const component = mount(
		<Router client={fetch(data)}>
			<CompletedItemsCard processId="35315" />
		</Router>
	);

	expect(component).toMatchSnapshot();
});

test('Should render component with failure state', () => {
	const component = mount(
		<Router client={fetchFailure()}>
			<CompletedItemsCard processId="35315" />
		</Router>
	);

	expect(component).toMatchSnapshot();
});