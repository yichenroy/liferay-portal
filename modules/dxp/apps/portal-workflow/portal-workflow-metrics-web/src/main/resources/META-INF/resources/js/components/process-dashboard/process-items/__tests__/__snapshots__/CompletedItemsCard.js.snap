// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render component 1`] = `
<MockRouter
  client={
    Object {
      "delete": [Function],
      "get": [Function],
      "post": [Function],
      "put": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "match": Object {
            "params": Object {
              "page": 1,
              "search": undefined,
              "sort": undefined,
            },
          },
          "pathname": "/processes/1/10/title%3Aasc",
          "search": "?backPath=%2F",
        },
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "match": Object {
                "params": Object {
                  "page": 1,
                  "search": undefined,
                  "sort": undefined,
                },
              },
              "pathname": "/processes/1/10/title%3Aasc",
              "search": "?backPath=%2F",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "match": Object {
              "params": Object {
                "page": 1,
                "search": undefined,
                "sort": undefined,
              },
            },
            "pathname": "/processes/1/10/title%3Aasc",
            "search": "?backPath=%2F",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <CompletedItemsCard
        processId="35315"
      >
        <ProcessItemsCard
          completed={true}
          description="Completed workflow items are counted and categorized by their SLA status."
          filter={
            Object {
              "defaultItem": null,
              "hideControl": true,
              "items": Array [],
              "key": "timeRange",
              "multiple": false,
              "name": "Completion Period",
            }
          }
          processId="35315"
          timeRange={null}
          title="Completed Items"
        >
          <Panel>
            <div
              className="container-fluid-1280 mt-4"
            >
              <div
                className="panel panel-secondary"
              >
                <Header
                  elementClasses={
                    Array [
                      "dashboard-panel-header",
                      "pb-0",
                    ]
                  }
                >
                  <div
                    className="panel-header dashboard-panel-header pb-0"
                  >
                    <div>
                      <div
                        className="autofit-row"
                      >
                        <div
                          className="autofit-col autofit-col-expand flex-row"
                        >
                          <span
                            className="mr-3"
                          >
                            Completed Items
                          </span>
                          <Tooltip
                            message="Completed workflow items are counted and categorized by their SLA status."
                            position="right"
                            width="288"
                          >
                            <span
                              className="workflow-tooltip"
                              onMouseLeave={[Function]}
                            >
                              <span
                                className="tooltip-trigger"
                                onMouseOver={[Function]}
                              >
                                <Icon
                                  iconName="question-circle-full"
                                >
                                  <svg
                                    className="lexicon-icon lexicon-icon-question-circle-full"
                                    focusable="false"
                                    role="presentation"
                                  >
                                    <use
                                      href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#question-circle-full"
                                    />
                                  </svg>
                                </Icon>
                              </span>
                            </span>
                          </Tooltip>
                        </div>
                        <div
                          className="autofit-col m-0 management-bar management-bar-light navbar"
                        >
                          <ul
                            className="navbar-nav"
                          >
                            <withRouter(Filter)
                              defaultItem={null}
                              filterKey="timeRange"
                              hideControl={true}
                              items={Array []}
                              key="timeRange"
                              multiple={false}
                              name="Completion Period"
                              position="right"
                            >
                              <Route>
                                <Filter
                                  defaultItem={null}
                                  filterKey="timeRange"
                                  hideControl={true}
                                  history={
                                    Object {
                                      "action": "POP",
                                      "block": [Function],
                                      "canGo": [Function],
                                      "createHref": [Function],
                                      "entries": Array [
                                        Object {
                                          "hash": "",
                                          "match": Object {
                                            "params": Object {
                                              "page": 1,
                                              "search": undefined,
                                              "sort": undefined,
                                            },
                                          },
                                          "pathname": "/processes/1/10/title%3Aasc",
                                          "search": "?backPath=%2F",
                                        },
                                      ],
                                      "go": [Function],
                                      "goBack": [Function],
                                      "goForward": [Function],
                                      "index": 0,
                                      "length": 1,
                                      "listen": [Function],
                                      "location": Object {
                                        "hash": "",
                                        "match": Object {
                                          "params": Object {
                                            "page": 1,
                                            "search": undefined,
                                            "sort": undefined,
                                          },
                                        },
                                        "pathname": "/processes/1/10/title%3Aasc",
                                        "search": "?backPath=%2F",
                                      },
                                      "push": [Function],
                                      "replace": [Function],
                                    }
                                  }
                                  items={Array []}
                                  location={
                                    Object {
                                      "hash": "",
                                      "match": Object {
                                        "params": Object {
                                          "page": 1,
                                          "search": undefined,
                                          "sort": undefined,
                                        },
                                      },
                                      "pathname": "/processes/1/10/title%3Aasc",
                                      "search": "?backPath=%2F",
                                    }
                                  }
                                  match={
                                    Object {
                                      "isExact": false,
                                      "params": Object {},
                                      "path": "/",
                                      "url": "/",
                                    }
                                  }
                                  multiple={false}
                                  name="Completion Period"
                                  position="right"
                                >
                                  <li
                                    className="dropdown nav-item"
                                  >
                                    <button
                                      aria-expanded={false}
                                      aria-haspopup="true"
                                      className="dropdown-toggle nav-link"
                                      onClick={[Function]}
                                      type="button"
                                    >
                                      <span
                                        className="mr-2 navbar-text-truncate"
                                      >
                                        Completion Period
                                      </span>
                                      <Icon
                                        iconName="caret-bottom"
                                      >
                                        <svg
                                          className="lexicon-icon lexicon-icon-caret-bottom"
                                          focusable="false"
                                          role="presentation"
                                        >
                                          <use
                                            href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#caret-bottom"
                                          />
                                        </svg>
                                      </Icon>
                                    </button>
                                    <div
                                      className="dropdown-menu dropdown-menu-right"
                                      role="menu"
                                    >
                                      <FilterSearch
                                        filteredItems={Array []}
                                        onChange={[Function]}
                                        totalCount={0}
                                      >
                                        <ul
                                          className="list-unstyled"
                                        >
                                          <li>
                                            <span
                                              className="disabled dropdown-item"
                                            >
                                              No results were found.
                                            </span>
                                          </li>
                                        </ul>
                                        <ul
                                          className="list-unstyled"
                                        />
                                      </FilterSearch>
                                    </div>
                                  </li>
                                </Filter>
                              </Route>
                            </withRouter(Filter)>
                          </ul>
                        </div>
                      </div>
                    </div>
                  </div>
                </Header>
                <Body>
                  <div
                    className="panel-body"
                  >
                    <div
                      className="pb-6 pt-5"
                    >
                      <LoadingState>
                        <span
                          aria-hidden="true"
                          className="loading-animation"
                        />
                      </LoadingState>
                    </div>
                  </div>
                </Body>
              </div>
            </div>
          </Panel>
        </ProcessItemsCard>
      </CompletedItemsCard>
    </Router>
  </MemoryRouter>
</MockRouter>
`;

exports[`Should render component with failure state 1`] = `
<MockRouter
  client={
    Object {
      "delete": [Function],
      "get": [Function],
      "post": [Function],
      "put": [Function],
    }
  }
>
  <MemoryRouter
    initialEntries={
      Array [
        Object {
          "match": Object {
            "params": Object {
              "page": 1,
              "search": undefined,
              "sort": undefined,
            },
          },
          "pathname": "/processes/1/10/title%3Aasc",
          "search": "?backPath=%2F",
        },
      ]
    }
    keyLength={0}
  >
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "canGo": [Function],
          "createHref": [Function],
          "entries": Array [
            Object {
              "hash": "",
              "match": Object {
                "params": Object {
                  "page": 1,
                  "search": undefined,
                  "sort": undefined,
                },
              },
              "pathname": "/processes/1/10/title%3Aasc",
              "search": "?backPath=%2F",
            },
          ],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "index": 0,
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "match": Object {
              "params": Object {
                "page": 1,
                "search": undefined,
                "sort": undefined,
              },
            },
            "pathname": "/processes/1/10/title%3Aasc",
            "search": "?backPath=%2F",
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <CompletedItemsCard
        processId="35315"
      >
        <ProcessItemsCard
          completed={true}
          description="Completed workflow items are counted and categorized by their SLA status."
          filter={
            Object {
              "defaultItem": null,
              "hideControl": true,
              "items": Array [],
              "key": "timeRange",
              "multiple": false,
              "name": "Completion Period",
            }
          }
          processId="35315"
          timeRange={null}
          title="Completed Items"
        >
          <Panel>
            <div
              className="container-fluid-1280 mt-4"
            >
              <div
                className="panel panel-secondary"
              >
                <Header
                  elementClasses={
                    Array [
                      "dashboard-panel-header",
                      "pb-0",
                    ]
                  }
                >
                  <div
                    className="panel-header dashboard-panel-header pb-0"
                  >
                    <div>
                      <div
                        className="autofit-row"
                      >
                        <div
                          className="autofit-col autofit-col-expand flex-row"
                        >
                          <span
                            className="mr-3"
                          >
                            Completed Items
                          </span>
                          <Tooltip
                            message="Completed workflow items are counted and categorized by their SLA status."
                            position="right"
                            width="288"
                          >
                            <span
                              className="workflow-tooltip"
                              onMouseLeave={[Function]}
                            >
                              <span
                                className="tooltip-trigger"
                                onMouseOver={[Function]}
                              >
                                <Icon
                                  iconName="question-circle-full"
                                >
                                  <svg
                                    className="lexicon-icon lexicon-icon-question-circle-full"
                                    focusable="false"
                                    role="presentation"
                                  >
                                    <use
                                      href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#question-circle-full"
                                    />
                                  </svg>
                                </Icon>
                              </span>
                            </span>
                          </Tooltip>
                        </div>
                        <div
                          className="autofit-col m-0 management-bar management-bar-light navbar"
                        >
                          <ul
                            className="navbar-nav"
                          >
                            <withRouter(Filter)
                              defaultItem={null}
                              filterKey="timeRange"
                              hideControl={true}
                              items={Array []}
                              key="timeRange"
                              multiple={false}
                              name="Completion Period"
                              position="right"
                            >
                              <Route>
                                <Filter
                                  defaultItem={null}
                                  filterKey="timeRange"
                                  hideControl={true}
                                  history={
                                    Object {
                                      "action": "POP",
                                      "block": [Function],
                                      "canGo": [Function],
                                      "createHref": [Function],
                                      "entries": Array [
                                        Object {
                                          "hash": "",
                                          "match": Object {
                                            "params": Object {
                                              "page": 1,
                                              "search": undefined,
                                              "sort": undefined,
                                            },
                                          },
                                          "pathname": "/processes/1/10/title%3Aasc",
                                          "search": "?backPath=%2F",
                                        },
                                      ],
                                      "go": [Function],
                                      "goBack": [Function],
                                      "goForward": [Function],
                                      "index": 0,
                                      "length": 1,
                                      "listen": [Function],
                                      "location": Object {
                                        "hash": "",
                                        "match": Object {
                                          "params": Object {
                                            "page": 1,
                                            "search": undefined,
                                            "sort": undefined,
                                          },
                                        },
                                        "pathname": "/processes/1/10/title%3Aasc",
                                        "search": "?backPath=%2F",
                                      },
                                      "push": [Function],
                                      "replace": [Function],
                                    }
                                  }
                                  items={Array []}
                                  location={
                                    Object {
                                      "hash": "",
                                      "match": Object {
                                        "params": Object {
                                          "page": 1,
                                          "search": undefined,
                                          "sort": undefined,
                                        },
                                      },
                                      "pathname": "/processes/1/10/title%3Aasc",
                                      "search": "?backPath=%2F",
                                    }
                                  }
                                  match={
                                    Object {
                                      "isExact": false,
                                      "params": Object {},
                                      "path": "/",
                                      "url": "/",
                                    }
                                  }
                                  multiple={false}
                                  name="Completion Period"
                                  position="right"
                                >
                                  <li
                                    className="dropdown nav-item"
                                  >
                                    <button
                                      aria-expanded={false}
                                      aria-haspopup="true"
                                      className="dropdown-toggle nav-link"
                                      onClick={[Function]}
                                      type="button"
                                    >
                                      <span
                                        className="mr-2 navbar-text-truncate"
                                      >
                                        Completion Period
                                      </span>
                                      <Icon
                                        iconName="caret-bottom"
                                      >
                                        <svg
                                          className="lexicon-icon lexicon-icon-caret-bottom"
                                          focusable="false"
                                          role="presentation"
                                        >
                                          <use
                                            href="http://localhost:8080/o/admin-theme/images/lexicon/icons.svg#caret-bottom"
                                          />
                                        </svg>
                                      </Icon>
                                    </button>
                                    <div
                                      className="dropdown-menu dropdown-menu-right"
                                      role="menu"
                                    >
                                      <FilterSearch
                                        filteredItems={Array []}
                                        onChange={[Function]}
                                        totalCount={0}
                                      >
                                        <ul
                                          className="list-unstyled"
                                        >
                                          <li>
                                            <span
                                              className="disabled dropdown-item"
                                            >
                                              No results were found.
                                            </span>
                                          </li>
                                        </ul>
                                        <ul
                                          className="list-unstyled"
                                        />
                                      </FilterSearch>
                                    </div>
                                  </li>
                                </Filter>
                              </Route>
                            </withRouter(Filter)>
                          </ul>
                        </div>
                      </div>
                    </div>
                  </div>
                </Header>
                <Body>
                  <div
                    className="panel-body"
                  >
                    <div
                      className="pb-6 pt-5"
                    >
                      <LoadingState>
                        <span
                          aria-hidden="true"
                          className="loading-animation"
                        />
                      </LoadingState>
                    </div>
                  </div>
                </Body>
              </div>
            </div>
          </Panel>
        </ProcessItemsCard>
      </CompletedItemsCard>
    </Router>
  </MemoryRouter>
</MockRouter>
`;
